lib/main-app.js

```diff
@@ -6,11 +6,19 @@ const path = require('path')
 const ChildProcess = require('child_process')
 const _ = require('lodash')
 const GhReleases = require('electron-gh-releases')
+const fs = require('fs')
+
 // electron.crashReporter.start()
 var ipcServer = null
 
 var mainWindow = null
 
+// support for portable mode
+app.setPath('userData',
+  fs.existsSync(path.join(path.dirname(process.execPath),
+    '.portable')) === true ? path.join(path.dirname(process.execPath),
+      'userdata') : app.getPath('userData'))
```

Comment:
It is too long line. Could you rewrite more clear?
Like this.

```javascript
const execPathDir = path.dirname(process.execPath);
const portableDirectoryExists = path.join(execPathDir, '.portable'));
const userDataPath = portableDirectoryExists ? path.join(execPathDir, 'userdata') : app.getPath('userData');
app.setPath('userData', userDataPath);
```

Reply:
In addtion, I think the directory name `.portable` is not best because it doesn't have information that it's of Boostnote in the name.
How about `.boost-portable` or anything like that?

Reply:
absolutely.. no problem at all. It is actually just a zero byte file and not a directory.

I'll clean up the code and change the file name. 

Thanks for considering the portable addition. Boostnote is an amazing tool for project note taking already and every time I look it gets better, thanks for all the work. I hope to contribute more when I can.